code/game/objects/structures/crates_lockers/closets/secure/craftable/freezer.dm

```diff
@@ -0,0 +1,204 @@
+
+/obj/structure/closet/secure_closet/freezer/empty/custom/CheckParts(list/parts_list)
+	for(var/obj/item/electronics/airlock/access_control in parts_list)
+		electronics = access_control
+		set_access()
+		electronics.moveToNullspace()
+		break
+
+/// copy over access of electronics
+/obj/structure/closet/secure_closet/freezer/empty/custom/proc/set_access()
+	if (electronics.one_access)
+		req_one_access = electronics.accesses
+	else
+		req_access = electronics.accesses
+
+/obj/structure/closet/secure_closet/freezer/empty/custom/multitool_act(mob/living/user, obj/item/tool)
+	if(locked)
+		balloon_alert(user, "unlock it first!")
+		return TRUE
+
+	if(!card_reader_installed)
+		balloon_alert(user, "requires card reader!")
+		return TRUE
+
+	access_locked = !access_locked
+	balloon_alert(user, "access panel [access_locked ? "locked" : "unlocked"]")
+	return TRUE
+
+/obj/structure/closet/secure_closet/freezer/empty/custom/attackby(obj/item/attacking_item, mob/living/user)
+	if(!broken && welded && istype(attacking_item, /obj/item/stock_parts/card_reader))
+		//you need to unlock to perform the operation else anyone can change access on a locked unit
+		if(card_reader_installed)
+			balloon_alert(user, "already installed!")
+			return TRUE
+
+		if(do_after(user, 4 SECONDS))
+			card_reader_installed = TRUE
+			attacking_item.moveToNullspace()
+			qdel(attacking_item)
+			balloon_alert(user, "card reader installed")
+
+		return TRUE
+	else if(istype(attacking_item, /obj/item/pen))
+		//you need to unlock to perform the operation else anyone can change name & description on a locked closet
+		if(locked)
+			balloon_alert(user, "unlock first!")
+			return TRUE
+
+		//you cant rename departmental lockers cause thats vandalism
+		if(access_type != PERSONAL)
+			balloon_alert(user, "not yours to rename!")
+			return TRUE
+
+		var/name_set = FALSE
+		var/desc_set = FALSE
+
+		var/str = tgui_input_text(user, "Personal Locker Name", "Locker Name")
+		if(!isnull(str))
+			name = str
+			name_set = TRUE
+
+		str = tgui_input_text(user, "Personal Locker Description", "Locker Description")
+		if(!isnull(str))
+			desc = str
+			desc_set = TRUE
+
+		var/bit_flag = NONE
+		if(name_set)
+			bit_flag |= UPDATE_NAME
+		if(desc_set)
+			bit_flag |= UPDATE_DESC
+		if(bit_flag != NONE)
+			update_appearance(bit_flag)
+
+		return TRUE
+	else if(card_reader_installed)
+		if(attacking_item.tool_behaviour == TOOL_CROWBAR)
+			if(!attacking_item.use_tool(src, user, 4 SECONDS))
+				return TRUE
+
+			new /obj/item/stock_parts/card_reader(drop_location())
+			card_reader_installed = FALSE
+			return TRUE
+
+		if(broken)
+			balloon_alert(user, "its broken!")
+			return TRUE
+		var/obj/item/card/id/id = null
+		if(istype(attacking_item, /obj/item/card/id))
+			id = attacking_item
+		else if(istype(attacking_item, /obj/item/modular_computer/pda))
+			var/obj/item/modular_computer/pda/pda = attacking_item
+			id = pda.computer_id_slot
```

Comment:
we have `GetID()` and you should be using that instead IMO

Reply:
works with wallets and stuff

Reply:
Actually we want to change access only when the player swipes their ID or an device containing an ID like the PDA. if we directly use GetID() then any interaction the player makes with the closet (like hitting it with a crowbar, or even opening it for example) would execute this action. That's why I specifically check the type of attacking_item I.e. what action are we doing on the closet with what item

Reply:
I think this is why it should be blueprints that make the closet instead of a closet that you amend after the fact